src/base/preferences.cpp

```diff
@@ -758,25 +759,32 @@ QString Preferences::getWebUIUsername() const
     return value<QString>(u"Preferences/WebUI/Username"_s, u"admin"_s);
 }
 
-void Preferences::setWebUIUsername(const QString &username)
+bool Preferences::setWebUIUsername(const QString &username)
 {
-    if (username == getWebUIUsername())
-        return;
+    if (username.length() < 3)
+        return false;
```

Comment:
I was hoping not to see this... this function should be moved to WebUI class. And I was hoping that Preferences class would only handle the save/load of values, not validating nor other data transforms.

Reply:
Reverted.
> this function (set/get webui username/password) should be moved to WebUI class.

I'll leave it to someone else. As I said above, I would like to limit myself to the necessary minimum of changes. This topic is outside the scope of my interests.
Now credentials are being checked in the UI layers (GUI and WebUI). As for WebAPI, it should first of all be redesigned in order to have a separate entry point for changing credentials, and not accept them together with the regular settings (yes, these are just regular settings from the GUI's point of view, but this is not the case for the WebUI).

Reply:
>this function (set/get webui username/password) should be moved to WebUI class

This will require deeper changes to the WebUI initialization process (I mean the case when we have to determine whether a temporary password is needed).

Reply:
>this function (set/get webui username/password) should be moved to WebUI class

Or be a throwaway lambda at the calling site...

>This will require deeper changes to the WebUI initialization process (I mean the case when we have to determine whether a temporary password is needed).

Personally, I don't require this for this PR.

Reply:
> > this function (set/get webui username/password) should be moved to WebUI class
> 
> This will require deeper changes to the WebUI initialization process (I mean the case when we have to determine whether a temporary password is needed).

Otherwise, Application will have to read the password hash from the Settings directly, bypassing the corresponding WebUI accessor.